### PR TITLE
Fix boolean views for HIP

### PR DESCRIPTION
**Pull request by emcastillo**
This will fix the following tests for hip:

```
cupy_tests/core_tests/test_ndarray_copy_and_view.py::TestArrayCopyAndView::test_astype_boolean_view
cupy_tests/core_tests/test_ndarray_elementwise_op.py::TestArrayElementwiseOp::test_add_array_boolarray
cupy_tests/core_tests/test_ndarray_elementwise_op.py::TestArrayElementwiseOp::test_iadd_array_boolarray
```

### CODE REVIEW

File: cupy/_core/_kernel.pyx [hunk 1]
```diff
@@ -969,7 +974,28 @@ cdef function.Function _get_ufunc_kernel(
     op.append(';')
     op.extend(out_op)
     operation = '\n'.join(op)
-
+    print(operation)
+    print('Operation done')
```
Comment (kmaehashi) [inline cupy/_core/_kernel.pyx:978]: Remove? 😉

Comment (emcastillo) [inline cupy/_core/_kernel.pyx:978]:
```suggestion
```

**Comment by kmaehashi** [comment]
For later reference - here is a reproducer (made by @emcastillo):

CUDA:
```py
>>> import cupy
>>> import numpy
>>> a = cupy.array([0, 1, 4], dtype=numpy.int8).view(dtype=numpy.bool_)
>>> print(a)
[False  True  True]
>>> print(a.astype(numpy.int8))
[0 1 1]
```

ROCm:

```py
>>> import cupy
>>> import numpy
>>> a = cupy.array([0, 1, 4], dtype=numpy.int8).view(dtype=numpy.bool_)
>>> print(a)
[False  True  True]
>>> print(a.astype(numpy.int8))
[0 1 4]
```

**Review by asi1024 — APPROVED** [review]

**Pull request merged**